src/cachemachine/rubinimageinfo.py

```diff
@@ -0,0 +1,346 @@
+import re
+from dataclasses import dataclass
+from enum import Enum, auto
+from typing import Optional
+
+import semver
+from docker_image import reference
+
+from .types import IncomparableImagesError
+
+DEFAULT_TAG = "latest"  # Docker convention
+DEFAULT_RECOMMENDED_TAG = "recommended"
+
+
+class RubinImageType(Enum):
+    """Class to capture the different kinds of image types possible in a
+    Rubin Lab Image container, given our standard tag set."""
+
+    RECOMMENDED = auto()
+    LATEST = auto()
+    LATEST_SOMETYPE = auto()
+    RELEASE = auto()
+    WEEKLY = auto()
+    DAILY = auto()
+    RELEASE_CANDIDATE = auto()
+    EXPERIMENTAL = auto()
+    MALFORMED_TAG = auto()
+
+
+# Helper functions that don't rely on any part of the class object and thus
+# are not methods.
+
+
+def _check_tag_is_lowercase(tag: str) -> bool:
```

Comment:
Why does this matter?  why not just lowercase everything on input?

Reply:
Because Docker tags, in general, can be mixed-case, and it is in fact legal, although a terrible idea, to have distinct tags differing only in case.  So I figured that the best way around this would be to claim that anything not strictly lower-case was a malformed tag, in which case it still shows up in the list-of-all-images, but is not parseable according to the Rubin Lab Tag rules...which I need to formalize and write down somewhere.